### ISSUE TITLE
Gcsfuse error 429 to many requests. 

### ISSUE THREAD

**Issue by oriavsapir** [GoogleCloudPlatform/gcsfuse#1709]
Hey, 

We’re trying to mount our buckets using the gcsfuse csi driver but we’re getting status code 429 upon heavy load (when trying to mount it on ~1000 pods in parallel)
The request is read only. 
--
compute: Received 429 `429 Too Many Requests
[13.2, 12:25] ‏{"error":"quota_exceeded","e │
│ rror_description":"The request was throttled due to rate limit. Please retry after a few seconds."}
---
Someone can solve the issue?
Thanks!

**Comment by shivannakarthik** [comment]
Have you referred to the below doc for best practices? Please see the rate limiting section
https://cloud.google.com/storage/docs/gcsfuse-performance-and-best-practices#rate_limiting

**Comment by oriavsapir** [comment]
> Have you referred to the below doc for best practices? Please see the rate limiting section https://cloud.google.com/storage/docs/gcsfuse-performance-and-best-practices#rate_limiting

Thanks for the response.
Yes, I see this doc.
But this is not good for me, and I want to ask if there is another way?

**Comment by raj-prince** [comment]
@oriavsapir do you see this as part of mount failure? Or this comes after the successful mount?

**Comment by gargnitingoogle** [comment]
This is a duplicate of internal bug id 325595226. I am trying to see how to mark/link it as such.

**Comment by gargnitingoogle** [comment]
Marking as P2 since this is a duplicate issue.

**Comment by gargnitingoogle** [comment]
@songjiaxun I assigned this issue to you as this is a duplicate of google internal bug id 325595226, which was identified as a GKE issue. Please close/update this one whenever you close that one.

**Comment by songjiaxun** [comment]
Hi @gargnitingoogle ,

According to the Google internal case, the 429 errors described in this issue was sporadic and not reproducible. The customer is not seeing this error any more.

The the customer reported another issue that was identified as a GKE issue. Now the new GKE version rollout completed. Starting from GKE 1.29.3-gke.1093000, the CSI driver injects the GCSFuse sidecar container as an init container. The Kubernetes native sidecar container feature manages the GCSFuse sidecar container lifecycle, which will solve the customer's issue.

To try out the new feature, please upgrade your GKE cluster to 1.29.3-gke.1093000 or later, and make sure ALL your nodes are also upgraded to GKE version 1.29 or later, then re-deploy your workloads.

Please feel free to close this issue for now. We will follow up with the customer for the GKE fixes.

**Comment by ashmeenkaur** [comment]
@oriavsapir Closing this issue based on https://github.com/GoogleCloudPlatform/gcsfuse/issues/1709#issuecomment-2041329644. Please feel free to re-open in case you still run into this issue with GKE cluster to 1.29.3-gke.1093000 or later.
